packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,358 @@
+
+                if (vertexIndex == 0.0) { 
+                    vBarycentric = vec3(0.0,1.0,0.0); 
+                }
+                else if (vertexIndex == 1.0) { 
+                    vBarycentric = vec3(1.0,0.0,0.0); 
+                }
+                else { 
+                    vBarycentric = vec3(0.0,0.0,1.0); 
+                }
+            `,
+            } : {
+            // Helper function definitions
+            "CUSTOM_FRAGMENT_DEFINITIONS": `
+                varying vec3 vBarycentric;
+
+                #if MODE == 1 || MODE == 2
+                    float edgeFactor(float thickness) {
+                        vec3 d = fwidth(vBarycentric);
+                        vec3 a3 = smoothstep(vec3(0.0), d * thickness, vBarycentric);
+                        return min(min(a3.x, a3.y), a3.z);
+                    }
+                #endif
+
+                #if (MODE == 3 && defined(UV1)) || (MODE == 4 && defined(UV2))
+                    float checkerboardFactor(vec2 uv) {
+                        float uIndex = floor(uv.x * uvScale);
+                        float vIndex = floor(uv.y * uvScale);
+                        return mod(uIndex + vIndex, 2.0);
+                    }
+                #endif
+            `,
+            // Channels that need lighting effects
+            // TODO: Find better place to update color pre-lighting instead of surfaceAlbedo
+            "CUSTOM_FRAGMENT_BEFORE_LIGHTS": `
+                #if MODE == 3 && defined(UV1)
+                    surfaceAlbedo = mix(checkerboardColor1, checkerboardColor2, checkerboardFactor(vMainUV1));
+                    #define DEBUG_BLEND
+                #elif MODE == 4 && defined(UV2)
+                    surfaceAlbedo = mix(checkerboardColor1, checkerboardColor2, checkerboardFactor(vMainUV2));
+                    #define DEBUG_BLEND
+                #elif MODE == 6
+                    surfaceAlbedo = materialColor;
+                    #define DEBUG_BLEND
+                #endif
+
+                vec3 newSurfaceAlbedo = surfaceAlbedo;
+            `,
+            // Channels that don't need lighting and don't blend with texture underneath
+            "CUSTOM_FRAGMENT_MAIN_END": `
+                #if MODE == 1
+                    gl_FragColor = mix(vec4(edgeColor,1.0), gl_FragColor, edgeFactor(edgeThickness));
+                #elif MODE == 2
+                    gl_FragColor = mix(vec4(edgeColor, 1.0), gl_FragColor, edgeFactor(dotEdgeThickness));
+                #elif MODE == 5 && defined(VERTEXCOLOR)
+                    gl_FragColor = vColor;
+                    #define DEBUG_BLEND
+                #endif
+
+                // Blending between select mesh & texture debug modes
+                #if DEBUGMODE > 0 && defined(DEBUG_BLEND)
+                    gl_FragColor = mix(gl_FragColor, vec4(newSurfaceAlbedo, 1.0), blendFactor);
+                #endif
+            `
+            // TODO: Add handling for when mesh data does not exist?

```

Comment:
Right now, the material will display normally if the data of the selected mesh debug mode is not available. 
I don't think this should be kept-- will elaborate on Wednesday.